Automattic/wp-calypso
# Checkout: Move WPLineItem and associates to own file

#### Changes proposed in this Pull Request

To help simplify some of the dependencies and to help eventually move some of this code outside of calypso (https://github.com/Automattic/wp-calypso/issues/52215 and https://github.com/Automattic/wp-calypso/pull/52890), this PR moves the `LineItem` and `NonProductLineItem` components into their own file.

It would be nice to move them directly to the `@automattic/wpcom-checkout` package, but they still have too many dependencies on calypso itself. Still, this will help to start untangling some of those dependencies.

As a related untangling, this also moves the line item `getLabel` and `getSublabel` functions (used only by `LineItem` currently) into the `@automattic/wpcom-checkout` package and removes the `isRenewal` function in favor of the already existing `isWpComProductRenewal` function.

![line-item](https://user-images.githubusercontent.com/2036909/130881785-fbd0f047-280d-4476-9f4d-8092a381bc2c.png)


#### Testing instructions

- Add a product to the cart and visit checkout.
- Verify that the products show up in checkout.
